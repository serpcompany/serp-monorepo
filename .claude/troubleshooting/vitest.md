https://vitest.dev/guide/common-errors.html