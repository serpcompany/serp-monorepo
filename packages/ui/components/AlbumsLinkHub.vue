<template>
  <div class="mt-2 grid grid-cols-1 gap-2 md:grid-cols-2 xl:grid-cols-2">
    <div v-for="album in albums" :key="album.slug">
      <nuxt-link :to="`/albums/${album.slug}/`" aria-label="album name">
        {{ album.name }} -
        <span v-for="artist in album.artists" :key="artist.slug">
          {{ artist.creditName }}
          <span v-if="artist.joinPhrase">{{ artist.joinPhrase }}</span>
        </span>
      </nuxt-link>
    </div>
  </div>
</template>

<script setup lang="ts">
import type { ReleaseGroupIndex } from '@serp/types/types'

defineProps({
  albums: {
    type: Array as () => ReleaseGroupIndex[],
    required: true
  }
})
</script>
