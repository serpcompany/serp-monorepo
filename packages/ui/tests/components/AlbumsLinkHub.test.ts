import { describe, expect, it } from 'vitest'
import AlbumsLinkHub from '../../components/AlbumsLinkHub.vue'
import ComponentRender from '../componentRender'

describe('albumsLinkHub Snapshot', () => {
  it.each([
    ['with no albums', { props: { albums: [] } }],
    [
      'with one album and one artist',
      {
        props: {
          albums: [
            {
              slug: 'album-1',
              name: 'Album One',
              artists: [{ slug: 'artist-1', credit_name: 'Artist One' }],
            },
          ],
        },
      },
    ],
    [
      'with multiple albums and multiple artists',
      {
        props: {
          albums: [
            {
              slug: 'album-1',
              name: 'Album One',
              artists: [
                {
                  slug: 'artist-1',
                  credit_name: 'Artist One',
                  join_phrase: ' & ',
                },
                { slug: 'artist-2', credit_name: 'Artist Two' },
              ],
            },
            {
              slug: 'album-2',
              name: 'Album Two',
              artists: [{ slug: 'artist-3', credit_name: 'Artist Three' }],
            },
          ],
        },
      },
    ],
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `AlbumsLinkHub ${desc}`,
        options,
        AlbumsLinkHub,
      )
      expect(html).toMatchSnapshot()
    },
  )
})
