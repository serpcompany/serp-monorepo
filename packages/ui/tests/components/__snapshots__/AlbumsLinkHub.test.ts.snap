// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlbumsLinkHub Snapshot > renders with multiple albums and multiple artists correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
">
  <div>
    <a
      aria-label="album name"
      href="/albums/album-1/"
    ><span>Album One - </span></a>
  </div>
  <div>
    <a
      aria-label="album name"
      href="/albums/album-2/"
    ><span>Album Two - </span></a>
  </div>
</div>
`;

exports[`AlbumsLinkHub Snapshot > renders with no albums correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
"></div>
`;

exports[`AlbumsLinkHub Snapshot > renders with one album and one artist correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
">
  <div>
    <a
      aria-label="album name"
      href="/albums/album-1/"
    ><span>Album One - </span></a>
  </div>
</div>
`;

exports[`albumsLinkHub Snapshot > renders with multiple albums and multiple artists correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
">
  <div>
    <a
      aria-label="album name"
      href="/albums/album-1/"
    ><span>Album One - </span></a>
  </div>
  <div>
    <a
      aria-label="album name"
      href="/albums/album-2/"
    ><span>Album Two - </span></a>
  </div>
</div>
`;

exports[`albumsLinkHub Snapshot > renders with no albums correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
"></div>
`;

exports[`albumsLinkHub Snapshot > renders with one album and one artist correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
">
  <div>
    <a
      aria-label="album name"
      href="/albums/album-1/"
    ><span>Album One - </span></a>
  </div>
</div>
`;
